src/rpc_client/src/meta_client.rs

```diff
@@ -275,16 +275,22 @@ impl MetaClient {
             .node
             .expect("AddWorkerNodeResponse::node is empty");
 
-        Ok((
-            Self {
-                worker_id: worker_node.id,
-                worker_type,
-                host_addr: addr.clone(),
-                inner: grpc_meta_client,
-                meta_config: meta_config.to_owned(),
-            },
-            system_params_resp.params.unwrap().into(),
-        ))
+        let meta_client = Self {
+            worker_id: worker_node.id,
+            worker_type,
+            host_addr: addr.clone(),
+            inner: grpc_meta_client,
+            meta_config: meta_config.to_owned(),
+        };
+
+        let meta_client_clone = meta_client.clone();
+        std::panic::update_hook(move |default_hook, info| {
```

Comment:
I'm afraid that `register_new` may be called multiple times in `playground` or `standalone` deployments. Consequently, the panic hook will also be updated multiple times, leading to duplicated events.

One possible solution is to use `std::sync::Once` here, but it may result in inaccurate reporting of the worker id and type. The ultimate solution might be making the behavior runtime-local.